instrumentation/internal/internal-class-loader/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/internal/classloader/LoadInjectedClassInstrumentation.java

```diff
@@ -30,7 +31,15 @@ public class LoadInjectedClassInstrumentation implements TypeInstrumentation {
 
   @Override
   public ElementMatcher<TypeDescription> typeMatcher() {
-    return extendsClass(named("java.lang.ClassLoader"));
+    // We explicitly exclude our own classloaders:
+    // The invokedynamic bootstrapping involves calling loadClass on those, which would cause
+    // an infinite recursion
+    return extendsClass(named("java.lang.ClassLoader"))
```

Comment:
If I understand it correctly, this mirrors the fact that `loadClass` is now called directly from `AgentClassLoader` and `InstrumentationModuleClassLoader` implementations, so maybe adding a comment to link to this instrumentation/method could help understand those two things are linked.